From 8dd758fefbac1f65218d13201ebda87e6ea7a3d0 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Thu, 29 Mar 2018 15:23:05 +0800
Subject: [PATCH 3549/5242] MLK-17940-1 gpu: imx: imx8_prg: Add prg_set_blit()
 helper support

commit  16cc274873c510acae653ab5987abf7ab7d61fef from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds prg_set_blit() helper support so that users may
set a particular PRG to be a part of a blit channel.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/imx8_prg.c    |   10 ++++++++++
 include/video/imx8-prefetch.h |    1 +
 2 files changed, 11 insertions(+)

diff --git a/drivers/gpu/imx/imx8_prg.c b/drivers/gpu/imx/imx8_prg.c
index 3d5d823..c236e16 100644
--- a/drivers/gpu/imx/imx8_prg.c
+++ b/drivers/gpu/imx/imx8_prg.c
@@ -73,6 +73,7 @@ struct prg {
 	struct clk *clk_apb;
 	struct clk *clk_rtram;
 	bool is_auxiliary;
+	bool is_blit;
 };
 
 static DEFINE_MUTEX(prg_list_mutex);
@@ -255,6 +256,15 @@ void prg_set_auxiliary(struct prg *prg)
 }
 EXPORT_SYMBOL_GPL(prg_set_auxiliary);
 
+void prg_set_blit(struct prg *prg)
+{
+	if (WARN_ON(!prg))
+		return;
+
+	prg->is_blit = true;
+}
+EXPORT_SYMBOL_GPL(prg_set_blit);
+
 struct prg *
 prg_lookup_by_phandle(struct device *dev, const char *name, int index)
 {
diff --git a/include/video/imx8-prefetch.h b/include/video/imx8-prefetch.h
index 849ef52..2a7c972a 100644
--- a/include/video/imx8-prefetch.h
+++ b/include/video/imx8-prefetch.h
@@ -42,6 +42,7 @@ void prg_configure(struct prg *prg, unsigned int width, unsigned int height,
 bool prg_stride_double_check(struct prg *prg,
 			     unsigned int stride, dma_addr_t baddr);
 void prg_set_auxiliary(struct prg *prg);
+void prg_set_blit(struct prg *prg);
 
 struct dprc;
 struct dprc *
-- 
1.7.9.5

